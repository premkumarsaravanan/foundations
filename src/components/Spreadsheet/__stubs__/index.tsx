import * as React from 'react'
import { Cell, SelectedMatrix } from '../types'
import ReactDataSheet from 'react-datasheet'

export const data: Cell[][] = [
  [
    { readOnly: true, value: 'Office Name' },
    { readOnly: true, value: 'Building Name' },
    { readOnly: true, value: 'Building No.' },
    { readOnly: true, value: 'Address 1' },
    { readOnly: true, value: 'Address 2' },
    { readOnly: true, value: 'Address 3' },
    { readOnly: true, value: 'Address 4' },
    { readOnly: true, value: 'Post Code' },
    { readOnly: true, value: 'Telephone' },
    { readOnly: true, value: 'Fax' },
    { readOnly: true, value: 'Email' }
  ],
  [
    { value: 'London' },
    { value: 'The White House' },
    { value: '15' },
    { value: 'London 1' },
    { value: '' },
    { value: 'Londom 3' },
    { value: '' },
    { value: 'EC12NH' },
    { value: '0845 0000' },
    { value: '' },
    { value: 'row1@gmail.com' }
  ],
  [
    { value: 'London2' },
    { value: 'The Black House' },
    { value: '11' },
    { value: '' },
    { value: '' },
    { value: 'Adress 3' },
    { value: '' },
    { value: 'EC12NH' },
    { value: '087 471 929' },
    { value: '' },
    { value: 'row2@gmail.com' }
  ],
  [
    { value: 'New York' },
    { value: 'Building A' },
    { value: '11' },
    { value: '' },
    { value: '' },
    { value: 'City Z' },
    { value: '' },
    { value: 'AL7187' },
    { value: '017 7162 9121' },
    { value: '' },
    { value: 'row3@gmail.com' }
  ]
]

export const cellRenderProps: ReactDataSheet.CellRendererProps<Cell> = {
  row: 3,
  col: 10,
  cell: { value: 'row3@gmail.com' },
  selected: false,
  editing: false,
  updated: false,
  attributesRenderer: jest.fn() as ReactDataSheet.AttributesRenderer<Cell>,
  className: 'cell',
  style: { background: 'red' },
  onMouseDown: jest.fn(),
  onMouseOver: jest.fn(),
  onDoubleClick: jest.fn(),
  onContextMenu: jest.fn(),
  children: <div>hi</div>
}

export const setData: React.Dispatch<Cell[][]> = jest.fn()

export const setSelected: React.Dispatch<SelectedMatrix> = jest.fn()

export const selectedMatrix = {
  start: { i: 0, j: 1 },
  end: { i: 2, j: 3 }
}
