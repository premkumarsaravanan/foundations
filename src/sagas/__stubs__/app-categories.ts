import { PagedResultCategoryModel_ } from '../../types/marketplace-api-schema'

export const appCategorieStub: PagedResultCategoryModel_ = {
  data: [
    {
      id: '3',
      name: 'Game',
      description: 'apps for game'
    },
    {
      id: '2',
      name: 'entertainment',
      description: 'apps for entertainment'
    },
    {
      id: '1',
      name: 'education',
      description: 'apps for education'
    }
  ],
  pageNumber: 1,
  pageSize: 15,
  pageCount: 3,
  totalCount: 3
}
