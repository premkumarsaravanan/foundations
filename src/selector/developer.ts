import { ReduxState } from '@/types/core'
import { oc } from 'ts-optchain'

export const selectDeveloperId = (state: ReduxState) => {
  return oc(state.auth.loginSession).loginIdentity.developerId()
}
