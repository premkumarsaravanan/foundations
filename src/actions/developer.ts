import { actionCreator } from '../utils/actions'
import ActionTypes from '../constants/action-types'
import { DeveloperItem } from '../reducers/developer'

export const developerRequestData = actionCreator<void>(ActionTypes.DEVELOPER_REQUEST_DATA)
export const developerLoading = actionCreator<boolean>(ActionTypes.DEVELOPER_LOADING)
export const developerReceiveData = actionCreator<DeveloperItem | undefined>(ActionTypes.DEVELOPER_RECEIVE_DATA)
export const developerClearData = actionCreator<null>(ActionTypes.DEVELOPER_CLEAR_DATA)
