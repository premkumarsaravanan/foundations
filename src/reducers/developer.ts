import { Action, FormState } from '../types/core'
import { isType } from '../utils/actions'
import {
  developerLoading,
  developerReceiveData,
  developerClearData,
  developerCreate,
  developerSetFormState
} from '../actions/developer'
import { PagedResultAppSummaryModel_, ScopeModel } from '@/types/marketplace-api-schema'

export interface DeveloperItem {
  data: PagedResultAppSummaryModel_
  scopes: ScopeModel[]
}

export interface DeveloperState {
  loading: boolean
  developerData: DeveloperItem | null
  formState: FormState
}

export const defaultState: DeveloperState = {
  loading: false,
  developerData: null,
  formState: 'PENDING'
}

const developerReducer = (state: DeveloperState = defaultState, action: Action<any>): DeveloperState => {
  if (isType(action, developerLoading)) {
    return {
      ...state,
      loading: action.data
    }
  }

  if (isType(action, developerReceiveData)) {
    return {
      ...state,
      loading: false,
      developerData: action.data || null
    }
  }

  if (isType(action, developerClearData)) {
    return {
      ...state,
      loading: false,
      developerData: action.data
    }
  }

  if (isType(action, developerSetFormState)) {
    return {
      ...state,
      formState: action.data
    }
  }

  return state
}

export default developerReducer
