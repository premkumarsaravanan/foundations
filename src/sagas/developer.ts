import fetcher from '../utils/fetcher'
import { URLS, PLATFORM_HEADERS, MARKETPLACE_HEADERS, REAPIT_API_BASE_URL } from '../constants/api'
import {
  developerLoading,
  developerReceiveData,
  developerSetFormState,
  developerRequestDataFailure
} from '../actions/developer'
import { put, fork, takeLatest, all, call } from '@redux-saga/core/effects'
import ActionTypes from '../constants/action-types'
import { errorThrownServer } from '../actions/error'
import errorMessages from '../constants/error-messages'
import { CreateDeveloperModel } from '../types/marketplace-api-schema'
import { Action } from '../types/core'
import { APPS_PER_PAGE } from '@/constants/paginator'

export const developerDataFetch = function*({ data: page }) {
  yield put(developerLoading(true))

  try {
    const response = yield call(fetcher, {
      url: `${URLS.apps}?PageNumber=${page}&PageSize=${APPS_PER_PAGE}`,
      method: 'GET',
      api: REAPIT_API_BASE_URL,
      headers: MARKETPLACE_HEADERS
    })
    if (response) {
      yield put(developerReceiveData({ data: response }))
    } else {
      yield put(developerRequestDataFailure())
    }
  } catch (err) {
    console.error(err.message)
    yield put(
      errorThrownServer({
        type: 'SERVER',
        message: errorMessages.DEFAULT_SERVER_ERROR
      })
    )
  }
}

export const developerCreate = function*({ data }: Action<CreateDeveloperModel>) {
  yield put(developerSetFormState('SUBMITTING'))

  try {
    const regResponse: true | undefined = yield call(fetcher, {
      url: URLS.developerCreate,
      api: REAPIT_API_BASE_URL,
      method: 'POST',
      body: data,
      headers: PLATFORM_HEADERS,
      isPrivate: false
    })
    const status = regResponse ? 'SUCCESS' : 'ERROR'
    yield put(developerSetFormState(status))
  } catch (err) {
    console.error(err.message)
    yield put(developerSetFormState('ERROR'))
    yield put(
      errorThrownServer({
        type: 'SERVER',
        message: errorMessages.DEFAULT_SERVER_ERROR
      })
    )
  }
}

export const developerRequestDataListen = function*() {
  yield takeLatest<Action<number>>(ActionTypes.DEVELOPER_REQUEST_DATA, developerDataFetch)
}

export const developerCreateListen = function*() {
  yield takeLatest(ActionTypes.DEVELOPER_CREATE, developerCreate)
}

const developerSagas = function*() {
  yield all([fork(developerRequestDataListen), fork(developerCreateListen)])
}

export default developerSagas
