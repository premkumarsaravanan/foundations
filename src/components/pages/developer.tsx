import * as React from 'react'
import { connect } from 'react-redux'
import { ReduxState } from '@/types/core'
import { authLogout } from '@/actions/auth'
import Loader from '@/components/ui/loader'
import ErrorBoundary from '@/components/hocs/error-boundary'
import { DeveloperState } from '@/reducers/developer'
import AppCard from '@/components/ui/app-card'

export interface DeveloperMappedActions {
  logout: () => void
}

export interface DeveloperMappedProps {
  developerState: DeveloperState
}
export type DeveloperProps = DeveloperMappedActions & DeveloperMappedProps

export const Developer: React.FunctionComponent<DeveloperProps> = ({ logout, developerState }) => (
  <div>
    <button onClick={() => logout()}>Logout</button>
    <div className="container">
      {developerState.loading ? (
        <Loader />
      ) : (
        <ErrorBoundary>
          <div className="row">
            {developerState.developerData &&
              developerState.developerData.data.map(child => (
                <div className="col-12 col-sm-6 col-md-4 col-lg-3 mb-4" key={child.id}>
                  <AppCard {...child} />
                </div>
              ))}
          </div>
        </ErrorBoundary>
      )}
    </div>
  </div>
)

const mapStateToProps = (state: ReduxState): DeveloperMappedProps => ({
  developerState: state.developer
})

const mapDispatchToProps = (dispatch: any): DeveloperMappedActions => ({
  logout: () => dispatch(authLogout())
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Developer)
