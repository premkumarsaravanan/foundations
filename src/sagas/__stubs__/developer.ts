import { DeveloperItem } from '../../reducers/developer'

export const developerDataStub: DeveloperItem = {
  data: [
    {
      id: '1',
      appName: 'string',
      developerName: 'string',
      developerId: 'string',
      displayImage: 'string',
      displayText: 'string'
    }
  ]
}
