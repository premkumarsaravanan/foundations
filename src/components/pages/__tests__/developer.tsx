import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { Developer, DeveloperProps } from '../developer'
import { developerDataStub } from '@/sagas/__stubs__/developer'

const props: DeveloperProps = {
  logout: jest.fn(),
  developerState: {
    loading: false,
    developerData: developerDataStub
  }
}

describe('Developer', () => {
  it('should match a snapshot', () => {
    expect(toJson(shallow(<Developer {...props} />))).toMatchSnapshot()
  })
})
