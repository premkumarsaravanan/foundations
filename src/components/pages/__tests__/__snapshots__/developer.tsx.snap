// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Developer should match a snapshot 1`] = `
<div
  className="container pt-5"
>
  <Connect(ErrorBoundary)>
    <div
      className="row"
    >
      <div
        className="col-12 col-sm-6 col-md-4 col-lg-3 mb-4"
        key="1"
      >
        <AppCard
          appName="string"
          developerId="string"
          developerName="string"
          displayImage="string"
          displayText="string"
          id="1"
        />
      </div>
    </div>
  </Connect(ErrorBoundary)>
</div>
`;
